aio/tools/transforms/angular-api-package/processors/mergeDecoratorDocs.js

```diff
@@ -109,18 +109,44 @@ module.exports = function mergeDecoratorDocs(log) {
 };
 
 function getInitializer(doc) {
-  var initializer = doc.symbol && doc.symbol.valueDeclaration && doc.symbol.valueDeclaration.initializer;
+  const declaration = doc.symbol && doc.symbol.valueDeclaration;
+  if (!declaration || !declaration.initializer) {
+    return;
+  }
+
+  let initializer = declaration.initializer;
+
   // There appear to be two forms of initializer:
-  //    export var Injectable: InjectableFactory =
-  //    <InjectableFactory>makeDecorator(InjectableMetadata);
+  //
+  // ```
+  // export const Injectable: InjectableFactory =
+  //       <InjectableFactory>makeDecorator(InjectableMetadata);
+  // ```
+  //
   // and
-  //    export var RouteConfig: (configs: RouteDefinition[]) => ClassDecorator =
-  //    makeDecorator(RouteConfigAnnotation);
-  // In the first case, the type assertion `<InjectableFactory>` causes the AST to contain an
-  // extra level of expression
-  // to hold the new type of the expression.
-  if (initializer && initializer.expression && initializer.expression.expression) {
+  //
+  // ```
+  // export const RouteConfig: (configs: RouteDefinition[]) => ClassDecorator =
+  //       makeDecorator(RouteConfigAnnotation);
+  // ```
+  //
+  // In the first case, the type assertion `<InjectableFactory>` causes the AST to contain an extra
+  // level of expression to hold the new type of the expression.
+  if (initializer.expression && initializer.expression.expression && initializer.expression.type) {
     initializer = initializer.expression;
   }
+
+  // It is also possible that the decorator call is wrapped in a call to `attachInjectFlag()`:
+  //
+  // ```
+  // const Optional: OptionalDecorator =
+  //       attachInjectFlag(makeParamDecorator('Optional'), InternalInjectFlags.Optional);
+  // ```
+  //
+  // If so, use the first argument of the call.
+  if (initializer && initializer.expression && initializer.expression.text === 'attachInjectFlag') {
```

Comment:
The `initializer` check should be redundant here, right?

```suggestion
  if (initializer.expression && initializer.expression.text === 'attachInjectFlag') {
```